Simplify and augment content (WIP) (#2420)
esl/MongooseIM: doc/user-guide/Getting-started.md

In this short guide we will set MongooseIM up and get users chatting right away.

> This will guide also you through basic operation, and validation.

Reviewer: ```suggestion
This will also walk you through basic operations, and validation.
```

> Training and getting to know MongooseIM is the goal here.

Reviewer: Alt:
In this short guide we will set MongooseIM up and get users chatting right away.
Our goal is to get to know MongooseIM, set it up, go through basic operations and validation.

Warning: This setup is not intended for production.

Note: This procedure has been tested on an Ubuntu 18.04.x LTS.

## Installation

> We recommand, you install mongooseIM binaries from a package Erlang Solutions delivers.

Reviewer 2: ~recommand~ -> recommend

Author: Thx

> Alternatively, check out our tutorial [How to build MongooseIM from source code](How-to-build.md) for an introduction to compiling, building and testing MongooseIM.

Reviewer: ```suggestion
Alternatively, check out our tutorial on [How to build MongooseIM from source code](How-to-build.md) for an introduction to compiling, building and testing MongooseIM.
```

### Download a package

Go to the [downloads](https://www.erlang-solutions.com/resources/download.html) section of the Erlang Solutions website, and choose the version of MongooseIM you want. The following sections describe the installation process for different operating systems.

### Ubuntu and Debian

Once the deb file is downloaded, open a terminal window and navigate to the directory containing the package. Use the following command to unpack and install MongooseIM:

```bash
sudo dpkg -i mongooseim_[version here].deb
```

### CentOS

An ODBC (RDBMS) driver must be installed on your machine to unpack and install from RPM packages. Enter the following command in a terminal window to install the latest unixODBC driver:
```bash
sudo yum install unixODBC
```
Once the RPM file is downloaded, open a terminal window and navigate to the directory containing the package. Use the following command to unpack and install MongooseIM:
```bash
sudo rpm -i mongooseim_[version here].rpm
```

## Running MongooseIM

> Warning: MongooseIM will use its default database Mnesia, which is faster and simpler to setup, but not intended for production purposes regarding persistent data.

Reviewer: ```suggestion
Warning: MongooseIM will use its default database - Mnesia, which is faster and simpler to setup, but not intended for production purposes when it comes to persistent data.
```

> Note: It is possible at anytime to use external databases, check at the end of this guide.

Reviewer: ```suggestion
Note: It is possible at anytime to use external databases. For more information see the end of this guide.
```

The following command will start the MongooseIM server:
```bash
mongooseimctl start
```

When you change the config file and want to restart the MongooseIM server:
```bash
mongooseimctl restart
```

Use the following command to stop the MongooseIM server:
```bash
mongooseimctl stop
```
This takes a few seconds.

At any given time, the following command shows the status of a MongooseIM server:
```bash
mongooseimctl status
```

> If the command replies `nodedown` then MongooseIM is not running. Else it will show its status `starting`, `started`, or `stopping`, and its version.

Reviewer: ```suggestion
If the command replies `nodedown` then MongooseIM is not running. In other cases it will show its status `starting`, `started`, or `stopping`, and its version.
```

When needed, you can also launch the server in the interactive mode:
```bash
mongooseimctl live
```
This will allow you to better detect and understand the errors in the configuration.

> When MongooseIM is properly running, the Erlang shell/console is then shown.

Reviewer: ```suggestion
When MongooseIM is properly running, the Erlang shell/console is shown.
```

> Just type twice Control-C to exit, the server will then be shut down.

Reviewer: ```suggestion
Just type Control-C twice to exit, the server will then be shut down.
```

For running MongooseIM in a non-interactive way within a supervision system (e.g. systemd), it is recommended to use the foreground mode:
```bash
mongooseimctl foreground
```
Typing Control-C will stop the server.

You can check server loglevel:
```bash
mongooseimctl get_loglevel
```

## Chat users

### Registering (creating) users

The default XMPP domain served by MongooseIM right after installation is `localhost`.

You can register (create) users with the `mongooseimctl` utility.

This command registers the user `user@domain` using password `password`.
```bash
mongooseimctl register_identified user domain password
```
Examples:
```bash
mongooseimctl register_identified alice localhost qwerty
mongooseimctl register_identified bob localhost 12345678
mongooseimctl register_identified carol localhost abc123
mongooseimctl register_identified dan localhost dan
```

> Warning: The password is entered manually on the command line, a history is accessible to the command line users. This method is not recommended for production use, you may prefer for example LDAP.

Reviewer: ```suggestion
Warning: The password is entered manually in the command line and history is accessible to the command line users. This method is not recommended for production use, you may prefer for example LDAP.
```

{[h,<<"localhost">>,module,mod_muc_light_commands],'FLAT'}.
{[h,<<"localhost">>,module,mod_muc],'FLAT'}.
{[h,<<"localhost">>,module_opt,mod_muc,host],"muc.@HOST@"}.
{[h,<<"localhost">>,module_opt,mod_muc,access],muc}.
{[h,<<"localhost">>,module_opt,mod_muc,access_create],muc_create}.
```

### Roster versioning

For faster contact list downloads at each client/app (re)connection, edit the configuration file:
```erlang
{mod_roster, [
              {versioning, true},
              {store_current_id, true}
             ]},
```
Verify with:
```bash
mongooseimctl print_flat_config | grep roster
 [{service_admin_extra,[{submods,[node,accounts,sessions,vcard,gdpr,roster,
{[h,<<"localhost">>,module,mod_roster],'FLAT'}.
{[h,<<"localhost">>,module_opt,mod_roster,versioning],true}.
{[h,<<"localhost">>,module_opt,mod_roster,store_current_id],true}.
```
### Review configuration

Read and double-check your configuration in a different way, after starting MongooseIM:
```bash
mongooseimctl print_flat_config
```

If MongooseIM does not start because the configuration file is broken in some way:
```bash
mongooseimctl live
```

## Using an XMPP/Jabber client/app

> The following steps uses the registered users on the MongooseIM server, done above.

Reviewer: ```suggestion
The following steps use the registered users on the MongooseIM server, done above.
```

Users that are registered on your server can now add their accounts in a chat application like Gajim (specifying either the server’s IP address or domain name), and start chatting!

### Connect Gajim

Gajim is available on Ubuntu, CentOS & Windows.

> Warning: Gajim has an obsolete UX. However, it is still well maintained, and has console that is extremely useful for debugging and testing/validation purposes at the XMPP protocol level.

Reviewer: ```suggestion
Warning: Gajim has an obsolete UX. However, it is still well maintained, and has a console that is extremely useful for debugging and testing/validation purposes at the XMPP protocol level.
```

From the MongooseIM command line:
```bash
mongooseimctl send_message_chat from to body
```
Examples:
```bash
mongooseimctl send_message_chat carol@localhost alice@localhost hello
mongooseimctl send_message_chat carol@localhost bob@localhost hi
```

### Group chats

Use `alice`'s account to create a groupchat `channel` on your `muc.localhost` service, and configure it by making it persistent. Invite `bob` and `carol`. From `bob`'s' and `carol`'s accounts, accept the invitation and join the `channel` groupchat. All three users exchange messages.

### Contact lists

Use `carol`'s account to add `alice` and `bob` to her contact list. Use `alice`'s and `bob`'s accounts accept those additions.

Verify on the MongooseIM server:
```
mongooseimctl get_roster user host
```
Examples:
```
mongooseimctl get_roster alice localhost
mongooseimctl get_roster bob localhost
mongooseimctl get_roster carol localhost
```

### Profile (vCard)

Edit `alice`'s profile (vCard) in Gajim: **Modify Account...**, then **Profile**, just set her **Name** to `Alice`.

Verify on the MongooseIM server:
```bash
mongooseimctl get_vcard alice localhost FN
```

## Summary

> Now you have the minimum knowledge: you know how to simply deploy MongooseIM, configure some basic features, check/verify a few useful items, validate it both on the client side and server side, and manipulate a few good practices.

Reviewer: ```suggestion
Now you have the minimum knowledge: you know how to deploy MongooseIM, configure some basic features, check/verify a few useful items, validate it both on the client and server side, and utilize a few good practices.
```
